geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/FrequencyZ3.scala

```diff
@@ -0,0 +1,144 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import java.util.Date
+
+import com.clearspring.analytics.stream.frequency.{CountMinSketch, RichCountMinSketch}
+import com.vividsolutions.jts.geom.Geometry
+import org.joda.time.{DateTime, DateTimeZone, Seconds, Weeks}
+import org.locationtech.geomesa.curve.Z3SFC
+import org.opengis.feature.simple.SimpleFeature
+
+/**
+  * Estimates frequency counts at scale. Tracks geometry and date attributes as a single value.
+  *
+  * @param geomIndex geometry attribute index in the sft
+  * @param dtgIndex date attribute index in the sft
+  * @param precision number of bits of z-index that will be used
+  * @param eps (epsilon) with probability at least @see confidence, estimates will be within eps * N
+  * @param confidence percent - with probability at least confidence, estimates will be within @see eps * N
+  */
+class FrequencyZ3(val geomIndex: Int,
+                  val dtgIndex: Int,
+                  val precision: Int,
+                  val eps: Double = 0.005,
+                  val confidence: Double = 0.95) extends Stat {
+
+  override type S = FrequencyZ3
+
+  private val mask = Frequency.getMask(precision)
+
+  private [stats] val sketches = scala.collection.mutable.Map.empty[Short, CountMinSketch]
```

Comment:
Max histogram size = 8 bytes * number of bins
We use variable length encoding, so counts up to ~2M will take 3 bytes or less (instead of 8).
I have been using a max bin size of 10000, which equates to ~29kb on disk for 1B records.
When storing, I'm using the 'splitByWeek' method - that way each entry will only be 29kb (assuming 1B records per week...), and when using them we only need to retrieve the weeks that we care about.
It looks like gdelt is 40M records - that would be ~16722 records per week, which would end up as ~10kb on disk - so 24MB...